src/components/TextArea/useTextArea.ts

```diff
@@ -0,0 +1,22 @@
+'use client';
+import { ChangeEvent, ChangeEventHandler, useCallback, useState } from 'react';
+
+export type UseTextAreaProps = {
+  onChange: ChangeEventHandler;
+  maxLength?: number;
+};
+
+export const useTextArea = ({ onChange, maxLength }: UseTextAreaProps) => {
+  const [textCount, setTextCount] = useState(0);
+
+  const onChangeHandler = useCallback(
+    (e: ChangeEvent<HTMLTextAreaElement>) => {
+      onChange && onChange(e);
+      e.target.value = e.target.value.slice(0, maxLength);
+      setTextCount(e.target.value.length);
```

Comment:
e에 대한 값을 바꾼 후에 onChange에 e object를 넘기는 게 순서에 더 맞을 것 같아요.
사실 결과적으로는 로직이 똑같지만 읽는 맥락에서요!

```suggestion
      e.target.value = e.target.value.slice(0, maxLength);
      onChange && onChange(e);
      setTextCount(e.target.value.length);
```